x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/evaluate_condition.ts

```diff
@@ -51,14 +52,18 @@ export const evaluateCondition = async ({
   compositeSize: number;
   filterQuery?: string;
   lookbackSize?: number;
+  startTime?: number;
 }): Promise<Record<string, ConditionResult>> => {
   const { comparator, warningComparator, metric, customMetric } = condition;
   let { threshold, warningThreshold } = condition;
 
+  const to = startTime ? moment(startTime) : moment();
+
   const timerange = {
-    to: Date.now(),
-    from: moment().subtract(condition.timeSize, condition.timeUnit).toDate().getTime(),
-    interval: condition.timeUnit,
```

Comment:
Interesting, does this mean it was only accepting whole time units before?

Reply:
For the interval it was using just the `condition.timeUnit` which resolves to `s|m|h|w|d` and ignores the size of the unit which would be a number.  If the customer put "FOR THE LAST 2 hours` the timerange would query the last 2 hours of data but the `date_histogram` (from the Snapshot API) would bucket every hour because `h` is a valid interval; the interval should be `2h`.  The customer thinks they are alerting on a value that covers the last 2 hours but it really only covers the last hour.